### PR TITLE
Parameters

### PR DESCRIPTION
**Pull request by YunwooLee94**
Define struct for parameters of prediction, planning, and problem_settings.

### CODE REVIEW

File: los_keeper/src/wrapper/wrapper.cc [hunk 1]
```diff
@@ -46,7 +49,18 @@ update : {
 }
 }
 
-Wrapper::Wrapper() { target_manager_.reset(new TargetManager3D); }
+Wrapper::Wrapper() {
+  // TODO: Parameter Assignment? (los_server <-> wrapper <-> (prediction and planning module))
```
Comment (YunwooLee94) [inline los_keeper/src/wrapper/wrapper.cc:53]: After merge, could you check this part?

File: los_keeper/include/los_keeper/bernstein_polynomial_utils/bernstein_polynomial_utils.h [hunk 2]
```diff
@@ -17,28 +20,45 @@ class BernsteinPoly {
...
+
   float GetTerminalValue() { return bernstein_coeff_[degree_]; };
+
   BernsteinPoly ElevateDegree(int m); // Change to higher degree (m)
   BernsteinPoly operator+(const BernsteinPoly &rhs_);
+
   BernsteinPoly operator-(const BernsteinPoly &rhs_);
+
   BernsteinPoly operator*(const BernsteinPoly &rhs_);
+
```
Comment (icsl-Jeon) [inline los_keeper/include/los_keeper/bernstein_polynomial_utils/bernstein_polynomial_utils.h:61]: Why did you put new line between functions??

File: los_keeper/include/los_keeper/type_manager/type_manager.h [hunk 3]
```diff
@@ -3,13 +3,47 @@
...
 #include "thread"
 #include <pcl/point_cloud.h>
 #include <pcl/point_types.h>
+
 namespace los_keeper {
 typedef pcl::PointXYZ PclPoint;
 typedef pcl::PointCloud<PclPoint> PclPointCloud;
+
+struct PredictionParam {
+  int num_sample;                // The number of primitives
```
Comment (icsl-Jeon) [inline los_keeper/include/los_keeper/type_manager/type_manager.h:17]: Can you modify them to have a default value?
int num_sample{0} for all params!

**Review by icsl-Jeon — COMMENTED** [review]
Overall, I could not focus on important changes as you put too many new lines. Why did you put new lines and mix them in this PR? I am just curious..

**Comment by icsl-Jeon** [comment]
In general, it is good practice to minimize the number of file changed.

File: los_keeper/include/los_keeper/target_manager/target_manager.h [hunk 4]
```diff
@@ -43,98 +37,150 @@ class TargetManager {
...
+
   virtual void CalculateCentroid();
+
   virtual void CalculateCentroidSubProcess(const int &target_id, const int &start_idx,
                                            const int &end_idx, pair<int, float> &min_dist);
 
 public:
   TargetManager();
 
+  explicit TargetManager(const PredictionParam &param);
```
Comment (icsl-Jeon) [inline los_keeper/include/los_keeper/target_manager/target_manager.h:71]: Nice!!!

File: los_keeper/include/los_keeper/bernstein_polynomial_utils/bernstein_polynomial_utils.h (hunk 2 above)
Comment (YunwooLee94) [reply los_keeper/include/los_keeper/bernstein_polynomial_utils/bernstein_polynomial_utils.h:61]: I didn't do it on purpose either. Why did it happen like that? I don't know the reason either. I'll remove the redundant line space for you.

File: los_keeper/include/los_keeper/type_manager/type_manager.h (hunk 3 above)
Comment (YunwooLee94) [reply los_keeper/include/los_keeper/type_manager/type_manager.h:17]: Okay!

File: los_keeper/src/wrapper/wrapper.cc (hunk 1 above)
Comment (icsl-Jeon) [reply los_keeper/src/wrapper/wrapper.cc:64]: I will handle this part!

**Comment by YunwooLee94** [comment]
Discard this branch.

**Pull request closed without merging**